### PR TITLE
FIX: optimze waypoint issue #138

### LINKED ISSUE #138: Golang Directions SDK does not optimize waypoints
The Directions API does not optimize waypoints when the request specifies Optimize: true.

The issue is that the golang SDK adds `optimize=true` as its own query parameter, when it should be part of the `waypoints` query parameter as `optimize:true|...`. 

Example of *working* URL from https://developers.google.com/maps/documentation/directions/intro#Waypoints:
```
https://maps.googleapis.com/maps/api/directions/json?origin=Adelaide,SA&destination=Adelaide,SA&waypoints=optimize:true|Barossa+Valley,SA|Clare,SA|Connawarra,SA|McLaren+Vale,SA&key=YOUR_API_KEY
```

The issue here is that https://github.com/googlemaps/google-maps-services-go/blob/master/directions.go#L91 adds optimize as a separate query parameter, rather than appending the waypoint string `"optimize:true"`. 

The result is a *broken* URL:
```
https://maps.googleapis.com/maps/api/directions/json?optimize=true&origin=Adelaide,SA&destination=Adelaide,SA&waypoints=Barossa+Valley,SA|Clare,SA|Connawarra,SA|McLaren+Vale,SA&key=YOUR_API_KEY
```

### PR DESCRIPTION
**Pull request by AlphaWong**
Fix #138 

# ~Minimum go version change ( from 1.5 to 1.10 )~
- ~using `strings.Builder` only support 1.10 +~

Please review my code

Best,
Alpha

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -3,8 +3,9 @@ go_import_path: googlemaps.github.io/maps
 
 # See https://github.com/travis-ci/gimme/blob/master/.testdata/binary-linux for known golang versions
 go:
-  - 1.6.x
-  - 1.7.x
-  - 1.8.x
-  - 1.9.x
+  - "1.10.x"
```
Comment (domesticmouse) [inline .travis.yml]: Nope, Go on GAE still supports Go 1.6 at this point.

**Comment by AlphaWong** [comment]
Sure let me change it `strings.Builder`

**Comment by AlphaWong** [comment]
Changed please review again ~

File: .travis.yml [hunk 2]
```diff
@@ -7,4 +7,9 @@ go:
   - 1.7.x
   - 1.8.x
   - 1.9.x
+  - "1.10.x"
```
Comment (domesticmouse) [inline .travis.yml]: Do you want to go ahead and make these consistent?

Comment (AlphaWong) [reply .travis.yml]: @domesticmouse 
Do u mean adding a double quote ?
I do not get it.

**Comment by domesticmouse** [comment]
All good, I'll clean it up. Thanks!

**Pull request merged**